### PR TITLE
Remove parameter DynFlags parameter from runActionWithContext

### PR DESCRIPTION
**Pull request by fendor**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -75,5 +75,5 @@ _build/
 stack*.yaml.lock
 shake.yaml.lock
 
-# ignore hie.yaml's for testdata 
-test/**/*.yaml
+# ignore hie.yaml
+hie.yaml
```
Comment (mpickering) [inline .gitignore:79]: Why is this here? Don't you need them to run tests?

File: hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs [hunk 2]
```diff
@@ -78,85 +78,76 @@ modifyCache f = do
...
   maybe (return ()) cacheCradle =<< (currentCradle <$> getModuleCache)
-
+  iniDynFlags <- GHC.getSessionDynFlags
   -- Now load the new cradle
   cradle <- liftIO $ findLocalCradle fp
   traceShowM cradle
+  -- set a new session
   liftIO (GHC.newHscEnv iniDynFlags) >>= GHC.setSession
   liftIO $ setCurrentDirectory (BIOS.cradleRootDir cradle)
+  -- initilise the session
```
Comment (mpickering) [inline hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs:129]: typo

File: hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs [hunk 3]
```diff
@@ -78,85 +78,76 @@ modifyCache f = do
...
   setCurrentCradle crd
   return (IdeResultOk ())
 
-loadCradle iniDynFlags (NewCradle fp) = do
+loadCradle  (NewCradle fp) = do
   traceShowM ("New cradle" :: String, fp)
   -- Cache the existing cradle
   maybe (return ()) cacheCradle =<< (currentCradle <$> getModuleCache)
-
+  iniDynFlags <- GHC.getSessionDynFlags
```
Comment (mpickering) [inline hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs:122]: This is surely not right. The `iniDynFlags` is not supposed to be a clean set of `DynFlags`.

**Review by mpickering — COMMENTED** [review]
I don't think the change to `iniDynFlags` is correct.

File: .gitignore (hunk 1 above)
Comment (fendor) [reply .gitignore:79]: they are generated, just like the `stack.yaml`.

File: hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs (hunk 3 above)
Comment (fendor) [reply hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs:122]: I see my mistake, this pr seems to be just wrong, then
